kythe/go/platform/kzip/kzip.go

```diff
@@ -170,8 +173,9 @@ func NewReader(r io.ReaderAt, size int64) (*Reader, error) {
 	}, nil
 }
 
-func (r *Reader) unitPath(digest string) string { return path.Join(r.root, "units", digest) }
-func (r *Reader) filePath(digest string) string { return path.Join(r.root, "files", digest) }
+func (r *Reader) unitPath(digest string) string   { return path.Join(r.root, "units", digest) }
+func (r *Reader) unitPbPath(digest string) string { return path.Join(r.root, "units.pb", digest) }
```

Comment:
I would detect and store the unit prefix along with `root` in `NewReader()`.  It might also be worthwhile to check that if both units and units.pb are present they have the same number of entries.